extensions/webdav/src/org/exist/webdav/MiltonDocument.java

```diff
@@ -250,96 +235,39 @@ public Long getContentLength() {
 
         if (existDocument.isXmlDocument()) {
             // XML document, exact size is not (directly) known)
+            if (isMacFinder || SIZE_METHOD.EXACT == propfindSizeMethod) {
 
-            if (isPropFind) {
-
-                // PROPFIND
-                // In this scensario the XML document is not actually
-                // downloaded, only the size needs to be known.
-                // This is the most expensive scenario
-
-                if (isMacFinder || SIZE_METHOD.EXACT == propfindSizeMethod) {
-
-                    // Returns the exact size, default behaviour for Finder,
-                    // or when set by a system property
-
-                    if (LOG.isDebugEnabled()) {
-                        LOG.debug(String.format("Serializing XML to /dev/null to determine size (%s) MacFinder=%s",
-                                resourceXmldbUri, isMacFinder));
-                    }
-
-                    // Stream document to '/dev/null' and count bytes
-                    ByteCountOutputStream counter = new ByteCountOutputStream();
-                    try {
-                        existDocument.stream(counter);
-
-                    } catch (Exception ex) {
-                        LOG.error(ex);
-                    }
-
-                    size = counter.getByteCount();
-
-
-                } else if (SIZE_METHOD.NULL == propfindSizeMethod) {
-
-                    // Returns size unknown. This is not supported
-                    // by MacOsX finder
-
-                    size = null;
-
+                // Returns the exact size, default behaviour for Finder,
+                // or when set by a system property
 
-                } else {
-                    // Returns the estimated document size. This is the
-                    // default value, but not suitable for MacOsX Finder.
-                    size = existDocument.getContentLength();
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug(String.format("Serializing XML to /dev/null to determine size (%s) MacFinder=%s",
+                            resourceXmldbUri, isMacFinder));
                 }
 
+                // Stream document to '/dev/null' and count bytes
+                final ByteCountOutputStream counter = new ByteCountOutputStream();
+                try {
```

Comment:
WHat about putting this stream as try-resource and put `size = counter.getByteCount();` in the try-catch ?

Reply:
Done.